### PR TITLE
add secp256k1-internal-0.3

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by mseri** [comment]
Can you have a look at:
```

#=== ERROR while compiling secp256k1-internal.0.3.0 ===========================#
# context              2.1.2 | linux/s390x | ocaml-base-compiler.4.13.1 | pinned(https://gitlab.com/nomadic-labs/ocaml-secp256k1-internal/-/archive/v0.2/ocaml-secp256k1-internal-0.3.tar.bz2)
# path                 ~/.opam/4.13/.opam-switch/build/secp256k1-internal.0.3.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p secp256k1-internal -j 3
# exit-code            1
# env-file             ~/.opam/log/secp256k1-internal-6597-a5237d.env
# output-file          ~/.opam/log/secp256k1-internal-6597-a5237d.out
### output ###
#         test alias test/runtest (exit 1)
# (cd _build/default/test && ./test.exe)
# Testing `secp256k1'.
# This run has ID `LDPQLLTS'.
# 
#   [OK]          Num               0   basic.
#   [OK]          Scalar            0   basic.
#   [OK]          External          0   signature_of_string.
#   [OK]          External          1   valid_signature.
#   [OK]          External          2   invalid_signature.
#   [OK]          External          3   public_module.
#   [OK]          External          4   pubkey_creation.
# > [FAIL]        External          5   sign.
#   [OK]          External          6   recover.
# 
# ┌──────────────────────────────────────────────────────────────────────────────┐
# │ [FAIL]        External          5   sign.                                    │
# └──────────────────────────────────────────────────────────────────────────────┘
# [exception] File "test/test.ml", line 100, characters 4-10: Assertion failed
#             Raised at Dune__exe__Test.External.test_sign in file "test/test.ml", line 100, characters 4-38
#             Called from Alcotest_engine__Core.Make.protect_test.(fun) in file "src/alcotest-engine/core.ml", line 180, characters 17-23
#             Called from Alcotest_engine__Monad.Identity.catch in file "src/alcotest-engine/monad.ml", line 24, characters 31-35
#             
# Logs saved to `~/.opam/4.13/.opam-switch/build/secp256k1-internal.0.3.0/_build/default/test/_build/_tests/secp256k1/External.005.output'.
#  ──────────────────────────────────────────────────────────────────────────────
# 
# Full test results in `~/.opam/4.13/.opam-switch/build/secp256k1-internal.0.3.0/_build/default/test/_build/_tests/secp256k1'.
# 1 failure! in 0.004s. 9 tests run.
```
See: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/3c33a70ebf2d22c3e3b23e6fd6df284848774b9b/variant/opam-2.1,extras,s390x,secp256k1-internal.0.3.0,tests

**Comment by hhugo** [comment]
cc @victor-dumitrescu 
@mseri, I don't know what to do with that. Is there an easy way to get a s390x system ?
should we declare the package as incompatible with s390x ?

File: packages/secp256k1-internal/secp256k1-internal.0.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "dune" {>= "2.0"}
+  "dune-configurator"
+  "cstruct" {>= "6.0.0"}
+  "bigstring" {>= "0.1.1"}
+  "conf-pkg-config"
+  "hex" {with-test & >= "1.4.0"}
+  "alcotest" {with-test}
+  "js_of_ocaml-compiler" { with-test & >= "3.11"}
+]
+
```
Comment (mseri) [inline packages/secp256k1-internal/secp256k1-internal.0.3.0/opam:27]:
```suggestion

available: arch != "s390x"

```

**Comment by mseri** [comment]
You can test it with docker if you want:
```
git clone --recursive "https://github.com/ocaml/opam-repository.git" && cd "opam-repository" && git fetch origin "refs/pull/20542/head" && git reset --hard 3c33a70e
git fetch origin master
git merge 80d7a1e3e5cfc5e27111bdbe8344b63862d7eda1
cat > ../Dockerfile <<'END-OF-DOCKERFILE'
FROM ocaml/opam:debian-11-ocaml-4.13@sha256:9b433980046ab3c3d30e0e8df08ddfb72d9b1c613652b7646718214828b78171
USER 1000:1000
WORKDIR /home/opam
RUN for pkg in $(opam pin list --short); do opam pin remove "$pkg"; done
RUN opam repository remove -a multicore || true
RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam
RUN opam init --reinit -ni
ENV OPAMDOWNLOADJOBS 1
ENV OPAMERRLOGLEN 0
ENV OPAMSOLVERTIMEOUT 500
ENV OPAMPRECISETRACKING 1
RUN rm -rf opam-repository/
COPY --chown=1000:1000 . opam-repository/
RUN opam repository set-url --strict default opam-repository/
RUN opam pin add -k version -yn secp256k1-internal.0.3.0 0.3.0
RUN opam update --depexts
RUN opam remove secp256k1-internal.0.3.0 && opam install --deps-only secp256k1-internal.0.3.0 && opam install -v secp256k1-internal.0.3.0; \
    res=$?; \
    test "$res" != 31 && exit "$res"; \
    export OPAMCLI=2.0; \
    build_dir=$(opam var prefix)/.opam-switch/build; \
    failed=$(ls "$build_dir"); \
    for pkg in $failed; do \
    if opam show -f x-ci-accept-failures: "$pkg" | grep -qF "\"debian-11\""; then \
    echo "A package failed and has been disabled for CI using the 'x-ci-accept-failures' field."; \
    fi; \
    done; \
    exit 1
RUN opam update --depexts
RUN opam remove secp256k1-internal.0.3.0 && opam install --deps-only --with-test secp256k1-internal.0.3.0 && opam install -v --with-test secp256k1-internal.0.3.0; \
    res=$?; \
    test "$res" != 31 && exit "$res"; \
    export OPAMCLI=2.0; \
    build_dir=$(opam var prefix)/.opam-switch/build; \
    failed=$(ls "$build_dir"); \
    for pkg in $failed; do \
    if opam show -f x-ci-accept-failures: "$pkg" | grep -qF "\"debian-11\""; then \
    echo "A package failed and has been disabled for CI using the 'x-ci-accept-failures' field."; \
    fi; \
    done; \
    exit 1

END-OF-DOCKERFILE
docker build -f ../Dockerfile .
```

**Comment by hhugo** [comment]
I looked into this. The two other releases also suffer from the same issue. I've added the constraint

**Comment by hhugo** [comment]
The current errors are for previous versions that I've only touched to add a constraint.

**Comment by mseri** [comment]
Looks fine, thanks

**Pull request merged**